### PR TITLE
Fix #858 Inconsistent empty splat assignment

### LINKED ISSUE #858: Inconsistent empty splat assignment
```livescript
[..., a, b] = [1]
[...x, y, z] = [1]
a is void and b is 1 and
x.length is 0 and y is 1 and z is void
```

Since line 2 is widely accepted as the correct behavior (i.e. splat cannot "unconsume") I don't see why line 1 shouldn't.

### PR DESCRIPTION
**Pull request by summivox**
See #858

TL;DR: `[a, ..., b, c, d] = [1 2 3]` should behave the same as `[a, ...m, b, c, d]`

### CODE REVIEW

File: src/ast.ls [hunk 1]
```diff
@@ -1660,10 +1660,14 @@ class exports.Assign extends Node
                     continue if skip and i+2 is len
                     start = i+1
                     @temps = [ivar = o.scope.temporary \i]
-                    val = if skip then node = Var ivar; Var val else
-                        Arr.wrap JS "#i < (#ivar = #val)
-                                  \ ? #{ util \slice }.call(#rite, #i, #ivar)
-                                  \ : (#ivar = #i, [])"
+                    val = switch
+                    | skip
```
Comment (vendethiel) [inline src/ast.ls]: Why did you change this from an if to a switch?

Comment (summivox) [inline src/ast.ls]: This patch is separated from summivox/LiveScript@e6591cfe35 . No particular reason.
Change it back to if?

**Comment by vendethiel** [comment]
Thanks!

**Comment by vendethiel** [comment]
Thanks!
